test/groovy/util/JenkinsWriteYamlRule.groovy

```diff
@@ -0,0 +1,53 @@
+package util
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+import org.yaml.snakeyaml.Yaml
+
+import static org.junit.Assert.*
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class JenkinsWriteYamlRule implements TestRule {
+
+    final BasePipelineTest testInstance
+    static final String YAML_DUMP = "YAML_DUMP" // key in files map to retrieve the string of an actual Yaml() dump.
+
+    Map files = [:]
+
+    JenkinsWriteYamlRule(BasePipelineTest testInstance) {
+        this.testInstance = testInstance
+    }
+
+    @Override
+    Statement apply(Statement base, Description description) {
+        return statement(base)
+    }
+
+    private Statement statement(final Statement base) {
+        return new Statement() {
+            @Override
+            void evaluate() throws Throwable {
+
+                testInstance.helper.registerAllowedMethod( 'writeYaml', [Map], { parameterMap ->
+                    assertNotNull(parameterMap.file)
+                    assertNotNull(parameterMap.data)
+
+                    Yaml yaml = new Yaml()
+                    StringWriter writer = new StringWriter()
+                    yaml.dump(parameterMap.data, writer)
+
+                    // Enable this to actually produce a file.
+                    // yaml.dump(parameterMap.data, new FileWriter(parameterMap.file))
+                    // yaml.dump(parameterMap.data, new FileWriter("test/resources/variableSubstitution/manifest_out.yml"))
+
+                    files[parameterMap.file] = parameterMap.data
+                    files[parameterMap.charset] = parameterMap.charset
+                    files[YAML_DUMP] = writer.toString()
+                })
```

Comment:
What about having one entry per file. Such a file entry contains file name, the file encoding, maybe the yml and the serialized form. `files[YAML_DUMP]` contains always the dump of the latest call. I would expect to have all dumps - in case of several invocations - available for performing checks at end of test.

Reply:
I tried to align this with the `JenkinsReadYamlRule`, but surely this could be done.

Reply:
**Update** - It's done. I pushed a new commit that resolves this.